File: db4oj/core/src/com/db4o/cs/messages/MReleaseSemaphore.java
/* Copyright (C) 2004   db4objects Inc.   http://www.db4o.com */

package com.db4o.cs.messages;

import com.db4o.*;
import com.db4o.cs.*;
import com.db4o.inside.*;

public final class MReleaseSemaphore extends MsgD {
	public final boolean processAtServer(ServerMessageDispatcher serverThread) {
		String name = readString();
		((LocalObjectContainer)stream()).releaseSemaphore(transaction(),name);
		return true;
	}
}

File: db4on/core/src/core/com/db4o/internal/cs/messages/MReleaseSemaphore.cs
namespace com.db4o.@internal.cs.messages
{
	public sealed class MReleaseSemaphore : com.db4o.@internal.cs.messages.MsgD
	{
		public sealed override bool ProcessAtServer(com.db4o.@internal.cs.ServerMessageDispatcher
			 serverThread)
		{
			string name = ReadString();
			((com.db4o.@internal.LocalObjectContainer)Stream()).ReleaseSemaphore(Transaction(
				), name);
			return true;
		}
	}
}
